### DISCUSSION TITLE
Why Prompt is removed? And How can i use prompt in version 6.12.1

### DISCUSSION THREAD

**Discussion by Rajgupta7080** [remix-run/react-router#10864] [Show and tell / tips]
I want to use prompt functionality

**Comment by brophdawg11** [comment]
These are available as 2 unstable hooks: [`unstable_useBlocker` and `unstable_usePrompt`](https://github.com/remix-run/react-router/issues/8139#issuecomment-1382428200).  We do not have them documented at the moment because they're unstable, but you can see an example of the blocker usage in https://github.com/remix-run/react-router/tree/main/examples/navigation-blocking.

`unstable_usePrompt` is of the format:

```jsx
unstable_usePrompt({ when: boolean, message: string });
```

**Comment by Rajgupta7080** [comment]
I want to know in which version prompt will be added.

**Reply by brophdawg11** [reply]
I'm not sure I understand.  

`unstable_usePrompt` exists and can be used today.  It's `unstable` because it doesn't have predicable behavior across browsers and can result in a messed up history stack.  For that reason, there an't plans at the moment to remove the `unstable` prefix.

There are no plans to add a `<Prompt>` component, you can build your own `unstable_usePrompt` if you want a componen t instead of a hook.
